app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -205,11 +207,15 @@ open class FenixApplication : Application() {
      * - https://github.com/mozilla/application-services/blob/master/docs/design/megazords.md
      * - https://mozilla.github.io/application-services/docs/applications/consuming-megazord-libraries.html
      */
-    private fun setupMegazord() {
-        // Note: This must be called as soon as possible
+    private fun setupMegazord(): Deferred<Unit> {
+        // Note: Megazord.init() must be called as soon as possible ...
         Megazord.init()
-        // This (and enabling RustLog) may be delayed if needed for performance reasons
-        RustHttpConfig.setClient(lazy { components.core.client })
+
+        return GlobalScope.async {
+            // ... but RustHttpConfig.setClient() and RustLog.enable() can be called later.
+            RustHttpConfig.setClient(lazy { components.core.client })
+            RustLog.enable()
```

Comment:
I wonder if you want to specify a different dispatcher here (e.g. `GlobalScope.async(Dispatchers.IO)`). Otherwise I think this will still happen on the same thread - just not sequentially?

Reply:
I am definitely not an expert on this, *but* according to the documentation, GlobalScope uses dispatchers that are [unconfined](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-global-scope/). That said, it does seem like a good idea to specify a launcher here that is explicitly off the the main thread. Do you agree?

Reply:
Colin has also asked that we explicitly specify the dispatcher. I agree with @pocmo that IO should be used here.

Reply:
Let me know what you think about the new version!